plutus-emulator/src/Wallet/Emulator/Types.hs

```diff
@@ -533,77 +514,23 @@ evalEmulated = \case
                 pure (Left err, txns)
     WalletRecvNotification wallet trigger -> fst <$> liftMockWallet wallet (handleNotifications trigger)
     BlockchainProcessPending -> do
-        emState <- get
+        initialState <- get
```

Comment:
It feels like a wart that we have to keep this around and update it separately to `finalState`, but I think it'll require some more work to avoid that.

Reply:
The `walletIndex` is a new addition. It seems like a new way of updating the Utxo index, which is kept in sync with the original way by an assertion. I do not know why this is happening, and I did not have the time to check. Having 2 ways of computing something seems redundant.

Reply:
The `walletIndex` has more information in it than the `UtxoIndex,` so we can't compute the former from the latter. It's annoying, but really it should be part of the wallet state, so should be updated when they receive their notifications. Future work.